app/models/configuration/DataSetConfiguration.scala

```diff
@@ -1,24 +1,40 @@
 package models.configuration
 
-import models.basics.BasicSettings
+import models.binary.DataSet
 import play.api.libs.json._
 
 case class DataSetConfiguration(configuration: Map[String, JsValue]) {
 
   def configurationOrDefaults = {
-    DataSetConfiguration.default.configuration ++ configuration
+    DataSetConfiguration.constructInitialDefault(List()).configuration ++ configuration
   }
 
 }
 
-object DataSetConfiguration extends BasicSettings {
+object DataSetConfiguration {
 
   implicit val dataSetConfigurationFormat = Json.format[DataSetConfiguration]
 
-  val default = DataSetConfiguration(
-    Map(
-      "fourBit" -> JsBoolean(false),
-      "quality" -> JsNumber(0),
-      "interpolation" -> JsBoolean(true),
-      "segmentationOpacity" -> JsNumber(20)))
+  def constructInitialDefault(dataSet: DataSet): DataSetConfiguration =
+    constructInitialDefault(dataSet.dataSource.toUsable.map(d => d.dataLayers.map(_.name)).getOrElse(List()))
+
+  def constructInitialDefault(layerNames: List[String]): DataSetConfiguration = {
+    val layerValues = Json.toJson(layerNames.map(layerName => Json.obj(layerName -> initialDefaultPerLayer)))
```

Comment:
@fm3 The layer format is off. It needs to be one map with layer names as keys, not an array of layers.